tests/session.test.ts

```diff
@@ -546,6 +546,86 @@ test.describe('Session', () => {
 		);
 	});
 
+	test('handles abort and navigation during completion', async ({ page }) => {
+		const aiMessage = page.locator('article', { hasText: 'Assistant' });
+		const errorMessage = page.getByText(
+			"Can't stop completion until the first response from Ollama is received"
+		);
+
+		async function setupAndRunPrompt() {
+			// Setup
```

Comment:
> As far as I understand, it should work, but it doesn't. It fails because, at line 609, it expects the completion to be started, but it is not. I think it has to be something about the stream response emulated at line 591.

I see some issues with the test but before I dig deeper we must fix the implementation because it will likely affect the behavior of the test.

That being said, we should probably do some cleanup here as well:
- `errorMessage` > `warningMessage`
- `"Can't stop completion until the first response from Ollama is received, please wait..."`

The fact that we need a `setupAndRunPrompt()` function feels like a code smell. A better way of handling this is to split the test into multiple smaller tests that are isolated from one-another, for example we could do:

```typescript
	test.describe('aborting completions', () => {
		test.beforeEach(async ({ page }) => {
			// move logic from setupAndRunPrompt() here
		});

		test("can't stop a while waiting for response", async ({ page }) => { ... });
		test('can stop a response in progress', async ({ page }) => { ... });
		test("can't navigate away from session while waiting for response", async ({ page }) => { ... });
		test('can navigate away from session while a response is in progress', async ({ page }) => { ... });
	});
```

Reply:
Also, whenever you run into weird issues with the tests you can install the [Playwright extension for VSCode](https://playwright.dev/docs/getting-started-vscode) and run it in debug mode: https://playwright.dev/docs/debug#run-in-debug-mode